vars/buildPlugin.groovy

```diff
@@ -61,7 +61,14 @@ def call(Map params = [:]) {
     }
 
     tasks[stageIdentifier] = {
+      int retryCounts = 1
       retry(count: 3, conditions: [kubernetesAgent(handleNonKubernetes: true), nonresumable()]) {
+        if (retryCounts >= 3 && platform != 'windows') {
+          // no spot instances for windows for now TODO change when available
+          // Use a spot instance for the 2 first try and nonspot for third and last
+          label = 'ubuntu-22-amd64-maven17-nonspot'
+        }
+        retryCounts = retryCounts + 1
```

Comment:
this is cleaner (and basically what I suggested before..)

count should start at 0 as its not being retried

```suggestion
      int retryCounter = 0
      retry(count: 3, conditions: [kubernetesAgent(handleNonKubernetes: true), nonresumable()]) {
        if (retryCounter > 1 && platform != 'windows') {
          // no spot instances for windows for now TODO change when available
          // Use a spot instance for the 2 first try and nonspot for third and last
          label = 'ubuntu-22-amd64-maven17-nonspot'
        }
        retryCounter++
```